crossandeancommands/make_nexus.py

```diff
@@ -0,0 +1,81 @@
+"""
+Create a nexus file from the most recent data on Edictor.
+"""
+from lingpy import Wordlist
+from lingpy.convert.strings import write_nexus
+from pathlib import Path
+
+
+def run(args):
+
+    wl = Wordlist.from_cldf(
+            Path("./", "cldf", "cldf-metadata.json").as_posix(),
+            # columns to be loaded from CLDF set
+            columns=(
+                "language_id",
+                "parameter_id",
+                "segments",
+                "language_subgroup",
+                "borrowing",
+                "cogid_cognateset_id"
```

Comment:
Do we export to `cognates.csv`? We have partial cognates in the end, so I doubt we have really handled this in our CLDF conversion?

Reply:
There it is:

https://github.com/lexibank/crossandean/blob/674d044d3471464b1d337bd398f16b42a04fc703/lexibank_crossandean.py#L113-L114

If you check this line, you see that this line will add all ZERO numbers as a distinct cogid to the CLDF, which is of course wrong. What needs to be done is, THIS line must be modified, by iterating over all `cogids`, and then only adding one, the first, which is not zero!

```python
                this_cogid = [x for x in cogids if x][0]:
                    args.writer.add_cognate(lexeme=lexeme, Cognateset_ID=this_cogid, Source=source)
```

This will guarantee your 0 annotation is accounted for. It does not filter borrowings, yet, but this is easy to do later. If you have two valid COGIDs, however, it'll take the first. This may not be what you want. If this IS the case, you need to modify:

```python
                this_cogid = "-".join([str(x) for x in cogids if x]):
                    args.writer.add_cognate(lexeme=lexeme, Cognateset_ID=this_cogid, Source=source)
```

This disambiguates, however, cases where you may want to assign the same cogid, but have two. So you may need to check for morphemes:

```python
                if morphemes:
                    new_cogids = []
                    for cogid, morpheme in cogids:
                        if not morpheme.startswith("_"):
                            new_cogids += [cogid]
                else:
                    new_cogids = cogids
                this_cogid = [x for x in new_cogids if x][0]:
                    args.writer.add_cognate(lexeme=lexeme, Cognateset_ID=this_cogid, Source=source)
```